### ISSUE TITLE
Trust Region Optimization within JAX

### ISSUE THREAD

**Issue by quattro** [google/jax#6320]
Would trust-region optimization have a home within JAX? I've been using an implementation of a jittable trust-ncg for my own purposes, [but recently integrated it within JAX framework ](https://github.com/quattro/jax/commit/f697495aac3574b44f78fadc179124a4dbe836c3) following the `minimize` API.

The implementation roughly follows that of the `trust-ncg` scipy implementation, but relies on gradients and hessian vector product functions computed using JAX. The current implementation is flexible enough to enable additional solvers for the constrained quadratic subproblem (similar to scipy for `trust-krylov`, `trust-dogleg`, etc.). It does not support PyTrees, but should not be too much trouble incorporating it later on.

Happy to initiate a pull request if there is broader interest.

**Comment by shoyer** [comment]
Hi @quattro, thanks for asking. I think `trust-ncg` would probably indeed be suitable for `jax.scipy.optimize`:

1. It's a standard method: it's implemented SciPy & recommended by Nocedal and Wright.
2. It's suitable for large-scale problems, so it might even be sensible to compose with neural networks.
3. It looks relatively straight-forward to implement. In particular, it avoids doing line-search, which is the tricky part of BFGS/L-BFGS (and I'm still not sure JAX does it right).

I'll ask a few other colleagues at Google to share their opinions, so maybe hold off a little bit before putting together a PR :)

**Comment by quattro** [comment]
Hi @shoyer, that's great to hear. Looking forward to the general consensus.

**Comment by MaAl13** [comment]
Hello, is there any update on this? I would like to use a jax version of trust-constr in scipy if this is existing.

**Comment by shoyer** [comment]
I'm afraid `scipy.optimize` is no longer considered within the scope of jax: 
https://jax.readthedocs.io/en/latest/jep/18137-numpy-scipy-scope.html#scipy-optimize

This might belong in JAXopt or Optimistix instead.
